tests/test_cluster.py

```diff
@@ -112,6 +113,28 @@ def test_cluster_server_statistics(sys_db, bad_db, cluster):
     assert err.value.error_code in {FORBIDDEN, DATABASE_NOT_FOUND}
 
 
+def test_cluster_server_maintenance_mode(sys_db, bad_db, cluster):
+    if not cluster:
+        pytest.skip("Only tested in a cluster setup")
+
+    server_id = sys_db.cluster.server_id()
+    result = sys_db.cluster.server_maintenance_mode(server_id)
+    assert result == {}
+
+    with assert_raises(ClusterMaintenanceModeError) as err:
+        bad_db.cluster.server_maintenance_mode(server_id)
+    assert err.value.error_code in {FORBIDDEN, DATABASE_NOT_FOUND}
+
+    sys_db.cluster.toggle_server_maintenance_mode(server_id, "maintenance", timeout=1)
```

Comment:
causing read timeout issues on _some_ of the cluster instances

Reply:
Tried to use a dbserver instead, but that didn't fix the issue.

Reply:
yeah really strange timeout, I don't have anything in mind yet

Reply:
thanks for the fix @apetenchea